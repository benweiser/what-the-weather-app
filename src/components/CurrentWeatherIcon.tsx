import React from 'react';
import Icon from './Icon';

export interface CurrentWeatherIconProps {
  currentConditions: number;
}

export const weather: { [key: number]: string } = {
  200: 'wi-thunderstorm',
  201: 'wi-thunderstorm',
  202: 'wi-thunderstorm',
  500: 'wi-day-rain',
  501: 'wi-day-rain',
  502: 'wi-day-rain',
  503: 'wi-day-rain',
  504: 'wi-day-rain',
  511: 'wi-day-showers',
  520: 'wi-day-showers',
  521: 'wi-day-showers',
  522: 'wi-day-showers',
  531: 'wi-day-showers',
  701: 'wi-raindrop',
  721: 'wi-day-haze',
  800: 'wi-day-sunny',
  801: 'wi-day-cloudy',
  802: 'wi-day-cloudy',
  803: 'wi-day-cloudy',
  804: 'wi-cloudy'
};

const CurrentWeatherIcon = ({ currentConditions }: CurrentWeatherIconProps) => {
  return (
    <Icon
      data-testid="CurrentWeatherIcon"
      size="xl"
      name={weather[currentConditions]}
    />
  );
};

export default CurrentWeatherIcon;
