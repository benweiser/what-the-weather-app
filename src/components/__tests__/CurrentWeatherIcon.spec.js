import renderer from "react-test-renderer";
import React from "react";
import CurrentWeatherIcon from "../CurrentWeatherIcon";

describe("Current Weather Icon", () => {
  test("renders", () => {
    const currentConditionCode = 500;
    const component = renderer.create(
      <CurrentWeatherIcon name={currentConditionCode} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
