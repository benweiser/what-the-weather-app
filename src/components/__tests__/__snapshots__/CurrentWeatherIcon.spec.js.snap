// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current Weather Icon renders 1`] = `
<span
  className="wi undefined css-tez4ss"
  size="xl"
/>
`;
